x-pack/plugins/stack_connectors/server/connector_types/gen_ai/gen_ai.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import type { AxiosError } from 'axios';
+import { GenAiRunActionParamsSchema } from '../../../common/gen_ai/schema';
+import type {
+  GenAiConfig,
+  GenAiSecrets,
+  GenAiRunActionParams,
+  GenAiRunActionResponse,
+} from '../../../common/gen_ai/types';
+import { GenAiBaseApiResponseSchema } from './api_schema';
+import { OpenAiProviderType, SUB_ACTION } from '../../../common/gen_ai/constants';
+
+export class GenAiConnector extends SubActionConnector<GenAiConfig, GenAiSecrets> {
+  private url;
+  private provider;
+  private key;
+
+  constructor(params: ServiceParams<GenAiConfig, GenAiSecrets>) {
+    super(params);
+
+    this.url = this.config.apiUrl;
+    this.provider = this.config.apiProvider;
+    this.key = this.secrets.apiKey;
+
+    this.registerSubActions();
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.RUN,
+      method: 'runApi',
+      schema: GenAiRunActionParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.TEST,
+      method: 'runApi',
+      schema: GenAiRunActionParamsSchema,
+    });
+  }
+
+  protected getResponseErrorMessage(error: AxiosError): string {
+    if (!error.response?.status) {
+      return 'Unknown API Error';
+    }
+    if (error.response.status === 401) {
+      return 'Unauthorized API Error';
+    }
+    return `API Error: ${error.response?.statusText}`;
+  }
+
+  public async runApi({ body }: GenAiRunActionParams): Promise<GenAiRunActionResponse> {
```

Comment:
I think the returned type should be `TypeOf<typeof GenAiBaseApiResponseSchema>`.

Reply:
`type GenAiRunActionResponse = TypeOf<typeof GenAiRunActionResponseSchema>;`

it is :)